src/client.rs

```diff
@@ -240,12 +240,10 @@ impl Client {
             }
         }
 
-        // Update the sync list
-        if let Ok(mut sync) = sync.write() {
-            *sync = synced_nodes;
-        }
-
-        network_info.write().map_err(|_| crate::Error::PoisonError)?.synced = true;
+        // Update the synced flag.
+        network_info.write().map_err(|_| crate::Error::PoisonError)?.synced = !synced_nodes.is_empty();
```

Comment:
Do we even want/need the synced field then or can we just check `!synced_nodes.is_empty()`?

Reply:
Good point